src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -1228,23 +1230,91 @@ class AbstractOption(models.Model):
     BOOLEAN = "boolean"
     DATE = "date"
 
+    SELECT = "select"
+    RADIO = "radio"
+    MULTI_SELECT = "multi_select"
+    CHECKBOX = "checkbox"
+
     TYPE_CHOICES = (
         (TEXT, _("Text")),
         (INTEGER, _("Integer")),
         (BOOLEAN, _("True / False")),
         (FLOAT, _("Float")),
         (DATE, _("Date")),
+        (SELECT, _("Select")),
+        (RADIO, _("Radio")),
+        (MULTI_SELECT, _("Multi select")),
+        (CHECKBOX, _("Checkbox")),
     )
 
+    empty_label = "------"
+    empty_radio_label = _("Skip this option")
+
     name = models.CharField(_("Name"), max_length=128, db_index=True)
     code = AutoSlugField(_("Code"), max_length=128, unique=True, populate_from='name')
     type = models.CharField(_("Type"), max_length=255, default=TEXT, choices=TYPE_CHOICES)
     required = models.BooleanField(_("Is this option required?"), default=False)
+    option_group = models.ForeignKey(
```

Comment:
I think it is worth adding validation to ensure this value is not populated unless the type is option or multi-option, and that it is not left empty otherwise.

Reply:
Will do

Reply:
done